File: src/main/java/uk/co/eelpieconsulting/feedlistener/rss/RssFeedItemBodyExtractor.java
package uk.co.eelpieconsulting.feedlistener.rss;

import java.util.List;

import org.apache.log4j.Logger;
import org.springframework.stereotype.Component;

import com.google.common.base.Strings;
import com.sun.syndication.feed.synd.SyndContentImpl;
import com.sun.syndication.feed.synd.SyndEntry;

@Component
public class RssFeedItemBodyExtractor {

	private static Logger log = Logger.getLogger(RssFeedItemBodyExtractor.class);
	
	@SuppressWarnings("unchecked")
	public String extractBody(final SyndEntry syndEntry) {
		final List<SyndContentImpl> contents = syndEntry.getContents();
		if (!contents.isEmpty()) {
			final String contentBody = contents.get(0).getValue();		
			return contentBody;
		}
		
		log.debug("No content body found; looking for description");
		String body = getItemDescription(syndEntry);
		if (!Strings.isNullOrEmpty(body)) {			
			return body;
		}
						
		return null;		
	}
	
	private String getItemDescription(final SyndEntry syndEntry) {
		return syndEntry.getDescription() != null ? syndEntry.getDescription().getValue() : null;
	}
	
}

File: src/main/kotlin/uk/co/eelpieconsulting/feedlistener/rss/RssFeedItemBodyExtractor.kt
package uk.co.eelpieconsulting.feedlistener.rss

import com.google.common.base.Strings
import com.sun.syndication.feed.synd.SyndContentImpl
import com.sun.syndication.feed.synd.SyndEntry
import org.apache.logging.log4j.LogManager
import org.springframework.stereotype.Component

@Component
class RssFeedItemBodyExtractor {

    private val log = LogManager.getLogger(RssFeedItemBodyExtractor::class.java)

    fun extractBody(syndEntry: SyndEntry): String? {
        val body = syndEntry.description?.value
        if (!Strings.isNullOrEmpty(body)) {
            return body
        }

        log.debug("No description found; looking for content")
        val contents = syndEntry.contents // contents returns an untyped list of SyndContentImpl
        if (contents.isNotEmpty()) {
            val firstItem = contents[0]
            if (firstItem != null && firstItem is SyndContentImpl) {
                return firstItem.value
            }
        }
        return null
    }

}